toolkits/slack/evals/eval_slack_chat.py

```diff
@@ -0,0 +1,625 @@
+            "List public channels and direct message conversations I am a member of",
+            list_public_channels_metadata,
+            list_direct_message_channels_metadata,
+        ),
+    ]
+
+    for name, user_message, *expect_called_tool_functions in cases:
+        suite.add_case(
+            name=name,
+            user_message=user_message,
+            expected_tool_calls=[
+                ExpectedToolCall(
+                    func=tool_function,
+                    args={},
+                )
+                for tool_function in expect_called_tool_functions
+            ],
+        )
+
+    return suite
+
+
+@tool_eval()
+def get_conversations_metadata_eval_suite() -> EvalSuite:
+    """Create an evaluation suite for tools getting conversations metadata."""
+    suite = EvalSuite(
+        name="Slack Tools Evaluation",
+        system_message="You are an AI assistant that can interact with Slack to send messages and get information from conversations, users, etc.",
+        catalog=catalog,
+        rubric=rubric,
+    )
+
+    suite.add_case(
+        name="Get conversation metadata by name",
+        user_message="Get the metadata of the #general channel",
+        expected_tool_calls=[
+            ExpectedToolCall(
+                func=get_conversation_metadata_by_name,
+                args={
+                    "conversation_name": "general",
+                },
+            ),
+        ],
+        critics=[
+            BinaryCritic(critic_field="conversation_name", weight=1.0),
+        ],
+    )
+
+    suite.add_case(
+        name="Get conversation metadata by id",
+        user_message="Get the metadata of the conversation with id '1234567890'",
+        expected_tool_calls=[
+            ExpectedToolCall(
+                func=get_conversation_metadata_by_id,
+                args={
+                    "conversation_id": "1234567890",
+                },
+            ),
+        ],
+        critics=[
+            BinaryCritic(critic_field="conversation_id", weight=1.0),
+        ],
+    )
+
+    return suite
+
+
+@tool_eval()
+def get_conversations_members_eval_suite() -> EvalSuite:
+    """Create an evaluation suite for tools getting conversations members."""
+    suite = EvalSuite(
+        name="Slack Tools Evaluation",
+        system_message="You are an AI assistant that can interact with Slack to send messages and get information from conversations, users, etc.",
+        catalog=catalog,
+        rubric=rubric,
+    )
+
+    user_messages = [
+        "Get the members of the #general channel",
+        "Get the members of the general channel",
+        "Get a list of people in the #general channel",
+        "Get a list of people in the general channel",
+        "Show me who's in the #general channel",
+        "Show me who's in the general channel",
+        "Who is in the #general channel?",
+        "Who is in the general channel?",
+    ]
+
+    for user_message in user_messages:
```

Comment:
@EricGustin use case for parameterized. I tend to think ``bulk_add_cases`` or ``add_case_batch`` would be good. arguments would accept input containers expected to be of the same length or create repeats to match passed container length if input is of 1 element